src/main/java/com/microsoft/sqlserver/jdbc/dtv.java

```diff
@@ -1083,8 +1083,16 @@ void execute(DTV dtv, Double doubleValue) throws SQLServerException {
 
         void execute(DTV dtv, BigDecimal bigDecimalValue) throws SQLServerException {
             if (DDC.exceedsMaxRPCDecimalPrecisionOrScale(bigDecimalValue)) {
-                String strValue = bigDecimalValue.toString();
-                tdsWriter.writeRPCStringUnicode(name, strValue, isOutParam, collation);
+                if (JDBCType.DECIMAL == dtv.getJdbcType() || JDBCType.NUMERIC == dtv.getJdbcType()) {
+                    // Throw exception for DECIMAL and NUMERIC Datatypes
+                    MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_valueOutOfRangeSQLType"));
+                    Object[] msgArgs = {JDBCType.DECIMAL == dtv.getJdbcType() ? SSType.DECIMAL : SSType.NUMERIC};
```

Comment:
Please also change
`Object[] msgArgs = {JDBCType.DECIMAL == dtv.getJdbcType() ? SSType.DECIMAL : SSType.NUMERIC};`
to 
`Object[] msgArgs = {JDBCType.DECIMAL == dtv.getJdbcType() ? JDBCType.DECIMAL : JDBCType.NUMERIC};`

Reply:
okay, I think this line can just be:
`Object[] msgArgs = {dtv.getJdbcType()};`